### DISCUSSION TITLE
Properly handle Too Many Requests error

### DISCUSSION THREAD

**Discussion by suXinjke** [RetroAchievements/RAIntegration#1117] [Ideas]
Right now you have to manually select 10-15 assets and upload them instead of being able to select them all, click the button and let the software do the work. It's extremely annoying when there are hundreds of assets to deal with, same goes to uploading code notes.

Please implement handling of HTTP 429 Too Many Requests, either RAIntegration itself should upload with delays, or it should wait for some time on it's own after receiving 429.

I also think we should be able to cancel the upload process in case upload gets stuck in this cycle of trying to avoid 429, due to low risk of faulty implementation.

**Comment by Jamiras** [comment]
The server limits API calls to 180 per minute (about 3 per second). You must be uploading a lot of stuff very fast to be getting throttled.

Personally, I've never encountered this issue, but I don't think I've tried uploading more than 100 or so things at once. (~50 achievements + badges). It might also matter than I'm about 8 timezones away from the server.

I agree that the software should automatically re-attempt 429 errors.
